Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -71,6 +72,9 @@ public PowerShellKernel() : base(DefaultKernelName)
         {
             _psHost = new PSKernelHost();
             _lazyPwsh = new Lazy<PowerShell>(CreatePowerShell);
+
+            var command = new SubmitCode(DollarProfileHelper.GetProfileScript());
+            DeferCommand(command);
```

Comment:
@colombod @jonsequitur I want to confirm... if I do this, will it not be handled lazily so it's only run when the user wan't to use a powershell notebook/cell?

Reply:
I want to add that `DeferCommand` is used [when initializing the FSharp kernel](https://github.com/dotnet/interactive/blob/master/Microsoft.DotNet.Interactive.FSharp/FSharpKernelExtensions.fs#L39).
I looked into the code, and `DeferCommand` [adds the code to `_deferredCommands`](https://github.com/dotnet/interactive/blob/master/Microsoft.DotNet.Interactive/KernelBase.cs#L72), which is a current queue object that is unique to every specific kernel object, so I think the deferred command will not activate the specific kernel until that kernel is really requested.
But I'd like @colombod and @jonsequitur to confirm :)

Reply:
Yes, deferred commands are queued and executed before the first use submission is processed.